tests/components/roon/test_config_flow.py

```diff
@@ -0,0 +1,181 @@
+    ) as mock_setup, patch(
+        "homeassistant.components.roon.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.1.1.1"}
+        )
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Roon Labs Music Player"
+    assert result2["data"] == {"host": "1.1.1.1", "api_key": "good_token"}
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_no_token(hass):
+    """Test we handle no token being returned (timeout or not authorized)."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    with patch("homeassistant.components.roon.config_flow.TIMEOUT", 0,), patch(
+        "homeassistant.components.roon.const.AUTHENTICATE_TIMEOUT", 0,
+    ), patch(
+        "homeassistant.components.roon.config_flow.RoonApi",
+        return_value=roonApiMock(None),
+    ):
+        await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.1.1.1"}
+        )
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.roon.config_flow.RoonApi", side_effect=CannotConnect,
+    ):
+        await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.1.1.1"}
+        )
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unknown_exception(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.roon.config_flow.RoonApi", side_effect=Exception,
+    ):
+        await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.1.1.1"}
+        )
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_form_host_already_exists(hass):
```

Comment:
What is this test meant to be testing? It looks like it's doing a successful entry creation.

Reply:
It's checking the happy path where there is an existing integration. Otherwise the code _might_ always reject a second integration.

Reply:
Right! I should have read the docstring.